src/platform/tls_schannel.c

```diff
@@ -2091,9 +2094,11 @@ CxPlatTlsWriteDataToSchannel(
                         "Indicate certificate received failed");
                     Result |= CXPLAT_TLS_RESULT_ERROR;
                     State->AlertCode = CXPLAT_TLS_ALERT_CODE_BAD_CERTIFICATE;
-                    break;
+                    goto IndicateReceivedCleanup;
                 }
 
+IndicateReceivedCleanup:
```

Comment:
Do we have other `goto` statements like this in the middle of a function? Usually `goto` is reserved for single-return style coding.

Reply:
We have some goto's for retrys in a few places. But there isn't a super great way to do this other then creating a leaf function. Which might be worth it TBH because this function is 1000 lines long.

Reply:
I decided to just make a leaf function, its cleaner.

Reply:
This is much more readable! Thank you